tests/integration/topic_log_test.go

```diff
@@ -0,0 +1,267 @@
+	case log.StringsType:
+		return zap.Strings(field.Key(), field.StringsValue())
+	case log.ErrorType:
+		return zap.Error(field.ErrorValue())
+	case log.StringerType:
+		return zap.Stringer(field.Key(), field.Stringer())
+	default:
+		return zap.Any(field.Key(), field.AnyValue())
+	}
+}
+
+func Fields(fields []log.Field) []zap.Field {
+	ff := make([]zap.Field, len(fields))
+	for i, f := range fields {
+		ff[i] = fieldToField(f)
+	}
+	return ff
+}
+
+//nolint:exhaustive // good enough.
+func Level(ctx context.Context) zapcore.Level {
+	switch log.LevelFromContext(ctx) {
+	case log.TRACE, log.DEBUG:
+		return zapcore.DebugLevel
+	case log.INFO:
+		return zapcore.InfoLevel
+	case log.WARN:
+		return zapcore.WarnLevel
+	case log.ERROR:
+		return zapcore.ErrorLevel
+	case log.FATAL:
+		return zapcore.FatalLevel
+	default:
+		return zapcore.InvalidLevel
+	}
+}
+
+func WithTraces(minLevel zapcore.Level, d trace.Detailer, opts ...log.Option) ydb.Option {
+	a := adapter{minLevel: minLevel}
+	return ydb.MergeOptions(
+		ydb.WithTraceDriver(log.Driver(a, d, opts...)),
+		ydb.WithTraceTable(log.Table(a, d, opts...)),
+		ydb.WithTraceScripting(log.Scripting(a, d, opts...)),
+		ydb.WithTraceScheme(log.Scheme(a, d, opts...)),
+		ydb.WithTraceCoordination(log.Coordination(a, d, opts...)),
+		ydb.WithTraceRatelimiter(log.Ratelimiter(a, d, opts...)),
+		ydb.WithTraceDiscovery(log.Discovery(a, d, opts...)),
+		ydb.WithTraceTopic(log.Topic(a, d, opts...)),
+		ydb.WithTraceDatabaseSQL(log.DatabaseSQL(a, d, opts...)),
+	)
+}
+
+type LogEntry struct {
+	Level       string `json:"L"`
+	Timestamp   string `json:"T"`
+	Namespace   string `json:"N"`
+	Message     string `json:"M"`
+	ContextName string `json:"context_name"`
+	Entity      string `json:"entity"`
+	Endpoint    string `json:"endpoint"`
+	Method      string `json:"method"`
+}
+
+func (l *LogEntry) FormatMessage() string {
+	return fmt.Sprintf(`"N":"%s","M":"%s","context_name":"%s","entity":"%s"`, l.Namespace, l.Message, l.ContextName, l.Entity)
+}
+
+func TestTopicReadMessagesLog(t *testing.T) {
+	_, buf := SetupZapLogger()
+
+	ctx := xtest.Context(t)
+
+	ctx = With(ctx, zap.String("context_name", "test_context"))
+	ctx = With(ctx, zap.String("entity", "reader"))
+
+	driver := connect(t, WithTraces(zapcore.DebugLevel, trace.DetailsAll))
+
+	db, reader := createFeedAndReaderForDB(ctx, t, driver, topicoptions.WithReaderLogContext(ctx))
+
+	sendCDCMessage(ctx, t, db)
+	msg, err := reader.ReadMessage(ctx)
+	require.NoError(t, err)
+	require.NotEmpty(t, msg.CreatedAt)
+	t.Logf("msg: %#v", msg)
+
+	require.NoError(t, err)
+	err = topicsugar.ReadMessageDataWithCallback(msg, func(data []byte) error {
+		t.Log("Content:", string(data))
+		return nil
+	})
+	require.NoError(t, err)
+
+	sendCDCMessage(ctx, t, db)
+	batch, err := reader.ReadMessagesBatch(ctx)
+	require.NoError(t, err)
+	require.NotEmpty(t, batch.Messages)
+
+	fileData, err := os.ReadFile("topic_log.json")
```

Comment:
We don't want to guarantee stable text of logs.

I think it is bad idea to canonization of the logs.